src/cluster.c

```diff
@@ -5227,18 +5227,22 @@ void createDumpPayload(rio *payload, robj *o, robj *key, int dbid) {
  * instance and that the checksum is ok.
  * If the DUMP payload looks valid C_OK is returned, otherwise C_ERR
  * is returned. */
-int verifyDumpPayload(unsigned char *p, size_t len) {
+int verifyDumpPayload(unsigned char *p, size_t len, uint16_t *rdbver_ptr) {
     unsigned char *footer;
     uint16_t rdbver;
     uint64_t crc;
 
+    if (!rdbver_ptr) {
+        rdbver_ptr = &rdbver;
+    }
```

Comment:
i think the more commonly used pattern is nicer and will also result in less changed lines.
i.e. all the code keeps using the stack based var, and we have an `if (rdbver_ptr) *rdbver_ptr = rdbver;` in one place.

Reply:
Fixed